db/migrate/20150114141626_actually_rename_smart_proxy_auth_related_settings.rb

```diff
@@ -0,0 +1,16 @@
+class ActuallyRenameSmartProxyAuthRelatedSettings < ActiveRecord::Migration
+  def up
+    old_restrict_registered = Setting[:restrict_registered_puppetmasters]
+    execute "DELETE FROM settings WHERE name='restrict_registered_puppetmasters'"
```

Comment:
You can also just fetch the record (`Setting.find_by_name`) and then call `destroy`, for a bit more consistency.